src/Network/NWDataTransferReport.cs

```diff
@@ -0,0 +1,174 @@
+			=> nw_data_transfer_report_get_sent_transport_byte_count (GetCheckedHandle (), pathIndex);
+
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern ulong nw_data_transfer_report_get_sent_transport_retransmitted_byte_count (OS_nw_data_transfer_report report, uint path_index);
+
+		public ulong GetTransportRetransmittedByteCount (uint pathIndex)
+			=> nw_data_transfer_report_get_sent_transport_retransmitted_byte_count (GetCheckedHandle (), pathIndex);
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern ulong nw_data_transfer_report_get_transport_smoothed_rtt_milliseconds (OS_nw_data_transfer_report report, uint path_index);
+
+		public ulong GetTransportSmootherRoundTripTime (uint pathIndex)
+			=> nw_data_transfer_report_get_transport_smoothed_rtt_milliseconds (GetCheckedHandle (), pathIndex);
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern ulong nw_data_transfer_report_get_transport_minimum_rtt_milliseconds (OS_nw_data_transfer_report report, uint path_index);
+
+		public ulong GetTransportMinimumRoundTripTime (uint pathIndex)
+			=> nw_data_transfer_report_get_transport_minimum_rtt_milliseconds (GetCheckedHandle (), pathIndex);
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern ulong nw_data_transfer_report_get_transport_rtt_variance (OS_nw_data_transfer_report report, uint path_index);
+
+		public ulong GetTransportMinimumRoundTripVariance (uint pathIndex)
+			=> nw_data_transfer_report_get_transport_rtt_variance (GetCheckedHandle (), pathIndex);
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern ulong nw_data_transfer_report_get_received_ip_packet_count (OS_nw_data_transfer_report report, uint path_index);
+
+		public ulong GetTransportReceivedIPPackageCount (uint pathIndex)
+			=> nw_data_transfer_report_get_received_ip_packet_count (GetCheckedHandle (), pathIndex);
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern ulong nw_data_transfer_report_get_sent_ip_packet_count (OS_nw_data_transfer_report report, uint path_index);
+
+		public ulong GetTransportSentIPPackageCount (uint pathIndex)
+			=> nw_data_transfer_report_get_sent_ip_packet_count (GetCheckedHandle (), pathIndex);
+
+		[DllImport (Constants.NetworkLibrary)]
+		unsafe static extern void nw_data_transfer_report_collect (OS_nw_data_transfer_report report, IntPtr queue, void *collect_block);
+
+		delegate void nw_data_transfer_report_collect_t (IntPtr block, IntPtr report);
+		static nw_data_transfer_report_collect_t static_CollectHandler = TrampolineCollectHandler;
+
+		[MonoPInvokeCallback (typeof (nw_data_transfer_report_collect_t))]
+		static void TrampolineCollectHandler (IntPtr block, IntPtr report)
+		{
+			var del = BlockLiteral.GetTarget<Action<NWDataTransferReport>> (block);
+			if (del != null) {
+				var nwReport = new NWDataTransferReport (report, owns: false);
+				del (nwReport);
+			}
+		}
+
+		public void Collect (DispatchQueue queue, Action<NWDataTransferReport> handler)
```

Comment:
`async`?

Reply:
@stephen-hawley I think I am able to write the full API with the async/await support and using completion sources, might be a good thing as a second pass for the API. @chamons can I have the time for that between Xcodes?